Birth and movement quick form blog posts (#16)
farmOS/farmOS-community-blog: posts/2023/movement-quick-form/index.md

checked.

For more information, see:
[farmOS Location Logic](https://farmOS.org/model/logic/location)

The Movement Quick Form streamlines this process by asking for only the
essential information and filling in the rest automatically.

The form starts by asking for the date when the movement took place. This
defaults to the current date, so you can record movements on the fly when they
happen. You can also adjust this date to record movements that happened in the
past, or movements that you are planning in the future. If you plan future
movements, be sure to uncheck the "Completed" box at the bottom of the form to
indicate that it hasn't happened yet.

Next, you must specify the assets that are moving. Start typing an asset's name
into the Assets field and select the one you want from the dropdown that
appears. Separate multiple assets with a comma.

You can select the locations that the assets are moving to in the same way. If
no locations are specified, this will clear the current location of the assets.

If the assets and locations you selected have a geometry already, they will
automatically show up on the map within the form. Current asset location

> geometry will be displayed in blue, and the location geometry will be displayed

Reviewer: ```suggestion
geometry will be displayed in blue, and the new geometry will be displayed
```

in orange.

> The location geometry is editable within the map, which allows you to customize

Reviewer: ```suggestion
The geometry for the movement is editable within the map, which allows you to customize
```

the asset's new geometry. Use the map's edit controls to select the geometry
you want to edit, and use the modify, move, or delete buttons to change them.
You can also use the polygon, line, point, and circle buttons to draw new
geometries.

Customizing the geometry of the movement does *not* change the geometry of the
location assets you selected. The new geometry is saved to the movement log
itself, which is what gets referenced when you look at the location of your
assets. Asset location geometry is determined by its movement logs. For more
information, see [Movements and location](https://farmos.org/guide/location/) in

> the farmOS user guide.

Reviewer: There's a lot of background packed in here which I think would be a little hard to parse for someone new to farmOS. I wonder if it would be better to simplify and lean more heavily on the official docs.

Something like...

```suggestion
Customizing the geometry of the movement does *not* change the geometry of the
location assets you selected. The custom geometry is only saved to the movement log
and thus only effects the moved asset(s). For more
information, see [Movements and location](https://farmos.org/guide/location/) in
the farmOS user guide.
```

<video autoplay control loop>
  <source src="./geometry.mp4" type="video/mp4">
</video>

One example where this is useful is moving animals with temporary fencing for
rotational grazing purposes. You may move a fence multiple times within the same
paddock, and you want to be able to record exactly where the animals were grazed
and for how long. The Movement Quick Form can accomplish this by referencing the
same location multiple times, but drawing the fence geometry differently each
time.

Another example could be moving a piece of equipment to a field, but specifying
the exact point where it was moved, so you (or someone else) can find it again
easily. Using the Movement Quick Form, you can specify a location like "North
Field" and also draw a point on the map exactly where the equipment is in "North
Field".

At the bottom of form there is an area to enter notes about the movement, which
will be saved to the "Notes" field of the log that is created.

The "Completed" checkbox at the bottom defaults to "Completed", but can be
unchecked if you are planning a future movement.

Upon submitting the form, you will see a message with a link to the new log that

> is created, and if you look at your asset records you will see that their
> current location and geometry has been updated.

Reviewer: ```suggestion
is created, and if you look at your asset records you will see that their
current location and geometry have been updated.
```

?
